en/physics-2d/physics-2d-system.md

```diff
@@ -0,0 +1,158 @@
+
+The rectangle test will test a specified rectangle in the world coordinate system, and if the bounding box of a collision body overlaps with this rectangle, then the collision body will be added to the return list.
+
+```ts
+const colliderList = PhysicsSystem2D.instance.testAABB(rect);
+```
+
+### Ray test
+
+The __Box2D__ physics module (not available in the Builtin module) also provides ray detection to detect which collision bodies a given line segment passes through. We can also obtain the normal vector at the point where the given line passes through and other useful information.
+
+```ts
+const results = PhysicsSystem2D.instance.rayCast(p1, p2, type, mask);
+
+for (const i = 0; i < results.length; i++) {
+    const result = results[i];
+    const collider = result.collider;
+    const point = result.point;
+    const normal = result.normal;
+    const fraction = result.fraction;
+}
+```
+
+The third parameter of the `rayCast` function specifies the type of detection, and the ray detection supports four types. This is because the ray detection of __Box2D__ is not detected from the nearest object of the ray starting point, so the result of the test can not guarantee that the result is sorted by the distance from the object near the start of the ray. __Cocos Creator__'s physics system will determine whether the __Box2d__ test results are sorted based on the type of detection. This type will affect the result return to user.
+
+- `ERaycast2DType.Any`
+
+  Detect any collider on the ray path. Once it detects any collider, it will immediately end the detection process and will no longer detect other objects.
+
+- `ERaycast2DType.Closest`
+
+  Detect the nearest collider on the ray path, which is the default for the `rayCast` detection, slightly slower than above method.
+
+- `ERaycast2DType.All`
+
+  Detect all colliders on the ray path, the order of the detected results is not fixed. In this type of detection, a collider may return multiple results because __Box2D__ run the detection by testing the fixture, and a collider may consist of multiple fixtures. This is a more costly method and will be slower than above methods.
+
+- `ERaycast2DType.AllClosest`
+
+  All colliders on the ray path are detected, but the return result is filtered and only the relevant information about the nearest point of each collider is returned, the slowest method of all.
+
+#### The result of ray detection
+
+The results of ray detection contain a lot of useful information, you can utilize these info according to the actual need.
+
+- __collider__
+
+  Specifies which collider the ray passes through.
+
+- __point__
+
+  Specifies the point at which the ray intersects the collider.
+
+- __normal__
+
+  Specifies the normal vector of the surface of the collider at the intersection.
+
+- __fraction__
+
+  Specifies the score of the intersection point at the ray.
+
+You can have a better understanding of the result of ray detection with the following figure.
```

Comment:
You can have a better understanding of the result of ray detection with the following figure.
-->
The following figure helps to better understand the result of ray detection.